src/plugins/kibana/public/dashboard/services/_saved_dashboard.js

```diff
@@ -46,16 +46,16 @@ module.factory('SavedDashboard', function (courier, config) {
 
   // if type:dashboard has no mapping, we push this mapping into ES
   SavedDashboard.mapping = {
-    title: 'string',
+    title: 'text',
```

Comment:
Note these mapping changes are going to break for anyone upgrading from an existing Kibana 4.x installation in which they have not saved at least one of each type of saved object. Eg, if they've saved visualizations and dashboards but never saved a search, saving a search will fail.